### PR TITLE
Addition (3rd_party_board): Add board support for Autosport Labs ESP32-CAN-X2 development board

### PR DESCRIPTION
**Pull request by brentpicasso**
----------
## Description of Change
Adds support for Autosport Labs ESP32-CAN-X2 development board

## Tests scenarios
Tested with the development board

## Related links
https://wiki.autosportlabs.com/ESP32-CAN-X2

### CODE REVIEW

File: variants/aslcanx2/pins_arduino.h
Comment (P-R-O-C-H-Y) [inline variants/aslcanx2/pins_arduino.h:12]: Those defines were moved from variants and are defined globally per SoC. Please remove it to fix warnings.
```suggestion
```

File: variants/aslcanx2/pins_arduino.h
Comment (P-R-O-C-H-Y) [inline variants/aslcanx2/pins_arduino.h:20]: Same here. Those defines were moved from variants and are defined globally per SoC. Please remove it to fix warnings.
```suggestion
```

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@brentpicasso Please take a look on my comments.

File: boards.txt
Comment (P-R-O-C-H-Y) [inline boards.txt:36478]: I recommend to change the build.board to be your board specific name, for example ASL_CAN_X2.

File: variants/aslcanx2/pins_arduino.h
Comment (brentpicasso) [reply variants/aslcanx2/pins_arduino.h]: thanks, I manually applied this change.

File: variants/aslcanx2/pins_arduino.h
Comment (brentpicasso) [reply variants/aslcanx2/pins_arduino.h]: thanks, I manually applied this change.

File: boards.txt
Comment (brentpicasso) [reply boards.txt:36478]: Thanks, I changed it to ASL-CAN-X2.

Comment (P-R-O-C-H-Y) [reply boards.txt:36478]: Please use underscores instead of dashes. Thanks!

Comment (brentpicasso) [reply boards.txt:36478]: Done! :) thank you.

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Comment by brentpicasso** [comment]
@P-R-O-C-H-Y  thank you for the quick approval.  Basic question - once this is merged, does it automatically show up in a nightly build of the Arduino IDE? Or is there a way existing IDE installations can get this update?  Googling didn't yield much info. Thanks again!

**Comment by P-R-O-C-H-Y** [comment]
> @P-R-O-C-H-Y thank you for the quick approval. Basic question - once this is merged, does it automatically show up in a nightly build of the Arduino IDE? Or is there a way existing IDE installations can get this update? Googling didn't yield much info. Thanks again!

You are welcome :) The nightly builds of Arduino IDE are unrelated to the ESP32 Arduino core.
Once we merge it there will be 2 ways to get the board to the Arduino IDE:
- Wait until next release 3.0.1 of the ESP32 Arduino core (this repo) and download it through the Boards manager in the IDE.
- Use manual installation of the ESP32 Arduino core (master branch) and get the boards immediately after it's merged to master. You can find the documentation of manual installation [HERE](https://docs.espressif.com/projects/arduino-esp32/en/latest/installing.html#windows-manual-installation), search for your OS :)

**Comment by lucasssvaz** [comment]
@brentpicasso For some reason the pre-commit bot is unable to push to your branch, maybe it could be related to organization permissions if your repo is in one. 

To fix that we can try enabling "Allow edits by maintainers" in this PR or alternatively, as it is a small PR, you can manually apply the changes shown [here](https://github.com/espressif/arduino-esp32/actions/runs/9270474998/job/25515801880?pr=9683)

**Comment by brentpicasso** [comment]
> @brentpicasso For some reason the pre-commit bot is unable to push to your branch, maybe it could be related to organization permissions if your repo is in one.
> 
> To fix that we can try enabling "Allow edits by maintainers" in this PR or alternatively, as it is a small PR, you can manually apply the changes shown [here](https://github.com/espressif/arduino-esp32/actions/runs/9270474998/job/25515801880?pr=9683)

Thanks, manually push the requested fix, including a correction to match product's PID / VID.

File: variants/aslcanx2/pins_arduino.h
Comment (lucasssvaz) [inline variants/aslcanx2/pins_arduino.h:11]:
```suggestion
#define BUILTIN_LED LED_BUILTIN  // backward compatibility
```

File: variants/aslcanx2/pins_arduino.h
Comment (lucasssvaz) [inline variants/aslcanx2/pins_arduino.h:22]:
```suggestion
```

**Review by lucasssvaz — CHANGES_REQUESTED** [review]
Just for the CI to pass

**Review by lucasssvaz — APPROVED** [review]

**Pull request merged**
